### PR TITLE
array-manipulation-1 | Rocky Truong

### PR DESCRIPTION
**Pull request by rocky-truong**
image

### CODE REVIEW

File: array-manipulation-1/src/initial.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/* exported initial */
+
+function initial(array) {
+  var newArray = [];
+  if (array.length === 0) {
+    return newArray;
+  }
```
Comment (codingcodymiller) [inline array-manipulation-1/src/initial.js:7]: This is not necessary here. If the array length is zero, the below loop would never start, so you get the same situation.

File: array-manipulation-1/src/reverse.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/* exported reverse */
+
+function reverse(array) {
+  var newArray = [];
+  if (array.length === 0) {
+    return newArray;
+  }
```
Comment (codingcodymiller) [inline array-manipulation-1/src/reverse.js:7]: This is not necessary here. If the array length is zero, the below loop would never start, so you get the same situation.

File: array-manipulation-1/src/tail.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* exported tail */
+
+function tail(array) {
+  var newArray = [];
+  if (array.length === 0) {
+    return newArray;
+  }
```
Comment (codingcodymiller) [inline array-manipulation-1/src/tail.js:7]: This is not necessary here. If the array length is zero, the below loop would never start, so you get the same situation.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by rocky-truong** [comment]
image

File: array-manipulation-1/src/compact.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/* exported compact */
+
+function compact(array) {
+  var newArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (array[i] !== false && array[i] !== null && Number.isNaN(array[i]) !== true && array[i] !== 0 && array[i] !== undefined && array[i] !== '') {
```
Comment (scbowler) [inline array-manipulation-1/src/compact.js:6]: There might be a simpler way to check for a falsy value... 🤔

**Review by scbowler — APPROVED** [review]
Great job implementing Cody's feedback, looks good! 👍 

Left a couple FYI comments below, no changes required

**Pull request merged**